app/Services/DeliverableService.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Services;
+
+use App\Resource;
+use App\Task;
+use App\Deliverable;
+use App\Contracts\DeliverableContract;
+
+class DeliverableService implements DeliverableContract
+{
+    public function getDataForViewingDeliverable($id)
+    {
+        $deliverable = Deliverable::find($id);
+        $tasks = $deliverable->getTasks()->get();
+        $show = false;
+        $taskId = null;
+        $data = [
+            'id' => $id,
+            'tasks' => $tasks,
+            'show' => $show,
+            'taskId' => $taskId,
+            'resourceName' => '',
+            'resourceTitle' => '',
+            'issues' => [],
+        ];
+
+        return $data;
+    }
+
+    public function getDataForViewingTask($did, $tid): array
+    {
+        $deliverable = Deliverable::find($did);
+        $tasks = $deliverable->getTasks()->get();
+        $show = true;
+        $taskId = $tid;
+
+        $task = Task::find($tid);
+        $resource = Resource::find($task->resource_id);
```

Comment:
Maybe use a different service resources?

Reply:
So this service does methods to do with the deliverable service, it needs all of those models in order for the method to work. Abstracting it further would be overkill that can create too many DB calls. Maybe I should do some renaming across the application to make things more clear